src/main/webapp/app/components/tokenInputGroups/TokenExtendButton.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { ReactNode } from 'react';
+import {
+  InputGroup,
+  Button,
+  Dropdown,
+  DropdownButton,
+  Row
+} from 'react-bootstrap';
+import classnames from 'classnames';
+import { observable, action } from 'mobx';
+import { observer } from 'mobx-react';
+import autobind from 'autobind-decorator';
+import { getMomentInstance } from 'app/shared/utils/Utils';
+import DayPickerInput from 'react-day-picker/DayPickerInput';
+import 'react-day-picker/lib/style.css';
+import moment from 'moment';
+import {
+  APP_LOCAL_DATETIME_FORMAT_Z_FORCE,
+  APP_LOCAL_DATE_FORMAT
+} from 'app/config/constants';
+import MomentLocaleUtils from 'react-day-picker/moment';
+import { DefaultTooltip } from 'cbioportal-frontend-commons';
+
+type TokenExtendButtonProps = {
+  currentDate: string;
+  afterChangeDate: (newDate: string) => void;
+};
+
+@observer
+export default class TokenExtendButton extends React.Component<
```

Comment:
After looking into the logics, I think this component is no longer needed to be a class. It could be a simple Functional component. Also the selectedDay no longer needs to be observable.